File: 2021/00/rust_template/benches/puzzle_benchmark.rs
use criterion::{black_box, criterion_group, criterion_main, Criterion};

// Puzzle input
const INPUT_FILENAME: &str = "input/puzzle_input.txt";

// Import puzzle solutions module
use rust_template;

/// Benchmark of part 1
fn task_1(c: &mut Criterion) {
  c.bench_function(
    "Day 0, Task 1: Rust Template", 
    |b| b.iter(|| rust_template::solution_1(
      black_box(&INPUT_FILENAME.to_string())
  )));
}

/// Benchmark of part 2
fn task_2(c: &mut Criterion) {
  c.bench_function(
    "Day 0, Task 2: Rust Template", 
    |b| b.iter(|| rust_template::solution_2(
      black_box(&INPUT_FILENAME.to_string())
  )));
}

criterion_group!(benches, task_1, task_2);
criterion_main!(benches);

File: 2023/00/kotlin_template/src/benchmarks/kotlin/aoc2023/day00/PuzzleBenchmark.kt
package aoc2023.day00

// Import solution and benchmarking annotations
import aoc2023.day00.kotlinTemplate.KotlinTemplate
import java.io.File
import org.openjdk.jmh.annotations.*

/** Puzzle input */
const val INPUT_FILENAME: String = "./input/puzzle_input.txt"

/** Benchmark framework */
@State(Scope.Benchmark)
open class PuzzleBenchmarks {
    val app = KotlinTemplate(File(INPUT_FILENAME).readText(Charsets.UTF_8))

    @Setup fun setUp() {}

    /** First task benchmark */
    @Benchmark
    fun task1() {
        app.solution1()
    }

    /** Second task benchmark */
    @Benchmark
    fun task2() {
        app.solution2()
    }
}
